components-api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala

```diff
@@ -38,12 +42,48 @@ trait Validator {
 
 //TODO: These validators should be moved to separated module
 
+case class SpelExpressionParameterValidator(
+    validationExpression: String,
+    validationFailedMessage: Option[String]
+) extends ParameterValidator {
+
+  private val parser: ExpressionParser   = new SpelExpressionParser()
+  private val parsedValidationExpression = parser.parseExpression(validationExpression)
+
+  override def isValid(paramName: String, value: Any, label: Option[String])(
+      implicit nodeId: NodeId
+  ): Validated[PartSubGraphCompilationError, Unit] = {
+    val context: StandardEvaluationContext = new StandardEvaluationContext()
+    context.setVariable(paramName, value)
+    Try(parsedValidationExpression.getValue(context, classOf[Boolean])).fold(
+      e =>
+        invalid(
+          CustomParameterValidationError(
+            s"Expression evaluation failed: ${e.getMessage}",
+            "Pleaase set valid expression",
+            paramName,
+            nodeId.id
+          )
+        ),
+      result => if (result) valid(()) else invalid(error(paramName, nodeId.id))
+    )
+  }
+
+  private def error(paramName: String, nodeId: String): CustomParameterValidationError = CustomParameterValidationError(
+    validationFailedMessage.getOrElse("This field has to satisfy the validation expression"),
```

Comment:
WDYT about displaying the expression in the fallback? I mean:
`Value '123' doesn't satisfy the defined validation expression '#value > 0'`. Also, shouldn't we have some placeholders in the validation failed message? E.g. `#value`? We can substitute it with just a `replaceAll`.

Reply:
done